services/update.py

```diff
@@ -81,6 +81,17 @@ def __init__(self, data, compat_mode='strict'):
         self.data['row'] = {}
         self.version_int = 0
         self.compat_mode = compat_mode
+        self.use_json = self.should_use_json()
+
+    def should_use_json(self):
+        # We serve JSON manifests to Firefox and Firefox for Android only,
+        # because we've seen issues with Seamonkey and Thunderbird.
+        # https://github.com/mozilla/addons-server/issues/7223
+        app = applications.APP_GUIDS.get(self.data.get('appID'))
+        if app and app.id in (applications.FIREFOX.id,
+                              applications.ANDROID.id):
+            return True
+        return False
```

Comment:
I'm pretty sure you can rewrite this `if` into a single `return` statement 😛

Reply:
3066801e7afd8d6353cf160a2fe880c229bbbc94